controllers/admin/AdminPayByLinkController.php

```diff
@@ -0,0 +1,224 @@
+<?php
+/**
+ * AltaPay module for PrestaShop
+ *
+ * Copyright © 2020 AltaPay. All rights reserved.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+class AdminPayByLinkController extends ModuleAdminController
```

Comment:
@BushraAsif Could you explain how this endpoint is being secured?

Reply:
This class is protected by Prestashop's default admin authentication as we are extending it from ModuleAdminController.

Reply:
@BushraAsif Just to confirm, all methods in this class will be protected by the admin authentication with username and password?

Please be careful to review the implementation of ModuleAdminController. I am afraid that this endpoints could be leveraged without auth, allowing an attacker to refund their own orders.

Reply:
We are extending ModuleAdminController which then extends AdminController. AdminController includes authorization checks to ensure that only authenticated users can access its methods.